packages/http-server-javascript/src/common/interface.ts

```diff
@@ -0,0 +1,270 @@
+  const opNameCase = parseCase(op.name);
+
+  const opName = opNameCase.camelCase;
+
+  const hasOptions = getAllProperties(op.parameters).some((p) => p.optional);
+
+  // const [successResult, _] = splitReturnType(ctx, op.returnType, module, opNameCase.pascalCase);
+
+  const returnTypeReference = emitTypeReference(ctx, op.returnType, op, module, {
+    altName: opNameCase.pascalCase + "Result",
+  });
+
+  const returnType = `Promise<${returnTypeReference}>`;
+
+  const params: string[] = [];
+
+  for (const param of getAllProperties(op.parameters)) {
+    // If the type is a value literal, then we consider it a _setting_ and not a parameter.
+    // This allows us to exclude metadata parameters (such as contentType) from the generated interface.
+    if (param.optional || isValueLiteralType(param.type)) continue;
+
+    const paramNameCase = parseCase(param.name);
+    const paramName = paramNameCase.camelCase;
+
+    const outputTypeReference = emitTypeReference(ctx, param.type, param, module, {
+      altName: opNameCase.pascalCase + paramNameCase.pascalCase,
+    });
+
+    params.push(`${paramName}: ${outputTypeReference}`);
+  }
+
+  const paramsDeclarationLine = params.join(", ");
+
+  yield* emitDocumentation(ctx, op);
+
+  if (hasOptions) {
+    const optionsTypeName = opNameCase.pascalCase + "Options";
+
+    emitOptionsType(ctx, op, module, optionsTypeName);
+
+    // TODO/witemple: how to extract parameter documentation?
+
+    const paramsFragment = params.length > 0 ? `${paramsDeclarationLine}, ` : "";
+
+    // prettier-ignore
+    yield `${opName}(ctx: Context, ${paramsFragment}options?: ${optionsTypeName}): ${returnType};`;
+    yield "";
+  } else {
+    // prettier-ignore
+    yield `${opName}(ctx: Context, ${paramsDeclarationLine}): ${returnType};`;
+    yield "";
+  }
+}
+
+/**
+ * Emit a declaration for an options type including the optional parameters of an operation.
+ *
+ * @param ctx - The emitter context.
+ * @param operation - The operation to emit the options type for.
+ * @param module - The module that the options type is written into.
+ * @param optionsTypeName - The name of the options type.
+ */
+export function emitOptionsType(
+  ctx: JsContext,
+  operation: Operation,
+  module: Module,
+  optionsTypeName: string
+) {
+  module.imports.push({
+    binder: [optionsTypeName],
+    from: ctx.syntheticModule,
+  });
+
+  const options = [...operation.parameters.properties.values()].filter((p) => p.optional);
+
+  ctx.syntheticModule.declarations.push([
+    `export interface ${optionsTypeName} {`,
+    ...options.flatMap((p) => [
+      `  ${parseCase(p.name).camelCase}?: ${emitTypeReference(ctx, p.type, p, module, {
+        altName: optionsTypeName + parseCase(p.name).pascalCase,
+      })};`,
+    ]),
+    "}",
+    "",
+  ]);
+}
+
+// TODO/witemple - everything below relating to split return types was inherited from the
```

Comment:
todo?

Reply:
I would like to look into this particular issue after merging. `SplitReturnType` is a concept that started from tsp-rust, and it's not exactly dead code in the JS emitter. More like "inoperative" I think. I don't want to go through the process of tearing this out just yet before merging and before I understand to what extent it affects the output. It was very important in tsp-rust and I had some ideas about how to leverage splitting unions  by success/error types in JS but I'm not sure about it yet.

Reply:
Tracking issue: https://github.com/microsoft/typespec/issues/3787